onnxscript/rewriter/ort_fusions/skip_normalization.py

```diff
@@ -2,115 +2,195 @@
-        return None
-    normalized, _mean, _inv_std_var, skip_sum = op.SkipSimplifiedLayerNormalization(
-        input,
-        skip,
-        gamma,
-        epsilon=epsilon,
-        _outputs=4,
-        _domain="com.microsoft",
-    )
-    return normalized, skip_sum
-
-
-_skip_rms_rule = pattern.RewriteRule(_skip_rms_norm_pattern, _skip_rms_normalization)
-
-skip_rms_normalization_rules = [_skip_rms_rule]
-skip_rms_normalization_ruleset = pattern.RewriteRuleSet(skip_rms_normalization_rules)
-
-
-def _skip_layer_norm_pattern(op, input, skip, gamma, beta, epsilon, stash_type):
-    skip_sum = op.Add(input, skip)
-    normalized = op.LayerNormalization(
-        skip_sum,
-        gamma,
-        beta,
-        axis=-1,
-        epsilon=epsilon,
-        stash_type=stash_type,
-    )
-    return normalized, skip_sum
-
-
-def _skip_layer_normalization(op, input, skip, gamma, beta, epsilon, stash_type):
-    if stash_type.value != 1:  # FLOAT type
-        return None
-    normalized, _mean, _inv_std_var, skip_sum = op.SkipLayerNormalization(
-        input,
-        skip,
-        gamma,
-        beta,
-        epsilon=epsilon,
-        _outputs=4,
-        _domain="com.microsoft",
-    )
-    return normalized, skip_sum
-
-
-# Fusion rule for Add + SkipLayerNormalization
-def _skip_layer_norm_add_bias_pattern(op, input, skip, gamma, beta, bias, epsilon, stash_type):
-    bias_sum = op.Add(input, bias)
-    normalized, _mean, _inv_std_var, skip_sum = op.SkipLayerNormalization(
-        bias_sum,
-        skip,
-        gamma,
-        beta,
-        epsilon=epsilon,
-        _outputs=4,
-        _domain="com.microsoft",
-    )
-    return normalized, skip_sum
-
-
-def _skip_layer_normalization_add_bias(
-    op, input, skip, gamma, beta, bias, epsilon, stash_type
-):
-    normalized, _mean, _inv_std_var, skip_sum = op.SkipLayerNormalization(
-        input,
-        skip,
-        gamma,
-        beta,
-        bias,
-        epsilon=epsilon,
-        _outputs=4,
-        _domain="com.microsoft",
-    )
-    return normalized, skip_sum
-
-
-_skip_layer_rule = pattern.RewriteRule(
-    _skip_layer_norm_pattern, _skip_layer_normalization, name="SkipLayerNorm"
+Dim = Union[int, ir.SymbolicDim]
+
+# Fusion rule for SkipRMSNormalization
+
+
+class SkipRmsNormFusion(pattern.RewriteRuleClassBase):
+    def __init__(self, name: str, has_bias: bool = False, bias_pre_add: bool = False):
+        """Fusion rule for SkipRMSNormalization."""
+        super().__init__(name=name)
+        self._has_bias = has_bias
+        self._bias_pre_add = bias_pre_add
+
+    def pattern(self, op, input, skip, gamma, bias, epsilon, stash_type):
+        if self._has_bias and self._bias_pre_add:
+            input = op.Add(input, bias)
+        skip_sum = op.Add(input, skip)
+        if self._has_bias and not self._bias_pre_add:
+            skip_sum = op.Add(skip_sum, bias)
+        normalized = op.SimplifiedLayerNormalization(
```

Comment:
Isn't this from com.microsoft domain? If so, we should specify that

Reply:
Ported over a similar note to what we have in RmsNormFusion